src/time_machine/cli.py

```diff
@@ -21,6 +22,40 @@
 DEDENT = "DEDENT"
 
 
+@dataclass
+class State:
+    """State collected during preprocessing to guide transformations."""
```

Comment:
please don't use a giant dataclass of state, I would prefer some local variables, which may use mini classes to collect several fields. Use plain classes, not dataclasses, for speed.